hnn_core/network_builder.py

```diff
@@ -538,6 +539,8 @@ def aggregate_data(self):
         _PC.allreduce(self._nrn_dipoles['L2_pyramidal'], 1)
         for nrn_arr in self._nrn_rec_arrays.values():
             _PC.allreduce(nrn_arr._nrn_voltages, 1)
+            # NB needed if multiple simulations are run in same python proc.
+            _CVODE.extra_scatter_gather_remove(nrn_arr._recording_callback)
```

Comment:
shouldn't this be in https://github.com/jonescompneurolab/hnn-core/blob/master/hnn_core/network_builder.py#L587

It would be nice to have everything related to "cleanup" in that function. I think Blake put that function for exactly the same reason you mention: "multiple simulations run in the same python proc."

Reply:
Yup, that's a better place for it.